sdk/messaging/azservicebus/go.mod

```diff
@@ -10,7 +10,6 @@ require (
 	github.com/Azure/go-amqp v0.17.0
 	github.com/devigned/tab v0.1.1
 	github.com/joho/godotenv v1.3.0
```

Comment:
This is a dev dep _only_ (ie, it's used for the tests). There's no distinction in here about it, but we don't use it in production code at all.

Reply:
It might be worth grouping test-only dependencies into a separate `require` clause with a preceding comment that says these are for tests only.  I believe that `go mod tidy` and other tools will preserve this structure (we should verify this before making the change).

Reply:
That's a great idea, I can at least try that.

I do wish there was a way to make it enforcing, but we'll go with this at least.

Reply:
It works! I updated it and split things a bit nicer.